workflowRDF/fairworkflow.py

```diff
@@ -1,21 +1,81 @@
+import os
 import rdflib
 from rdflib.namespace import RDF, RDFS, DC, XSD, OWL 
 import inspect
 from datetime import datetime
 
-PPLAN = rdflib.Namespace("http://purl.org/net/p-plan/")
-PLEX = rdflib.Namespace("https://plex.org/")
+PPLAN = rdflib.Namespace("http://purl.org/net/p-plan#")
 EDAM = rdflib.Namespace("http://edamontology.org/")
-PROV = rdflib.Namespace("http://www.w3.org/ns/prov/")
+PROV = rdflib.Namespace("http://www.w3.org/ns/prov#")
 DUL = rdflib.Namespace("http://ontologydesignpatterns.org/wiki/Ontology:DOLCE+DnS_Ultralite/")
 BPMN = rdflib.Namespace("https://www.omg.org/spec/BPMN/")
 PWO = rdflib.Namespace("http://purl.org/spar/pwo/")
 RDFG = rdflib.Namespace("http://www.w3.org/2004/03/trix/rdfg-1/")
 NP = rdflib.Namespace("http://www.nanopub.org/nschema#")
 
+
+def nanopublish(assertionrdf=None, uri=None):
+    """
+    Publish the given rdf as a nanopublication with the given URI
+    """
+
+    THISNP = rdflib.Namespace(uri+'#')
```

Comment:
It seems that example scripts in rdflib use all caps for namespaces (e.g. RDF, XSD, etc) so I was initially following that convention. I think it might be weird now, though, in this kind of use. I will make a change such that only standard ontology names are in all caps (such as PPLAN), and others are normal lower case python vars.